src/sentry/tasks/store.py

```diff
@@ -760,6 +769,16 @@ def _do_save_event(
             return
 
         try:
+            if killswitch_matches_context(
+                "store.load-shed-save-event-projects",
+                {
+                    "project_id": project_id,
+                    "event_type": event_type,
+                    "platform": data.get("platform") or "none",
+                },
+            ):
+                raise HashDiscarded("Load shedding save_event")
```

Comment:
To clarify, this will also leave the orphaned event payload in Redis?
If yes, let's add a comment about it for more visibility (can be fixed/improved later).

Reply:
it doesn't orphan, the exception is caught somewhere else again

we use exceptions as goto